validator/remote/src/main/java/tech/pegasys/teku/validator/remote/typedef/handlers/SendSignedBlockRequest.java

```diff
@@ -13,28 +13,73 @@
 
 package tech.pegasys.teku.validator.remote.typedef.handlers;
 
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_NOT_ACCEPTABLE;
 import static tech.pegasys.teku.validator.remote.apiclient.ValidatorApiMethod.SEND_SIGNED_BLINDED_BLOCK;
 import static tech.pegasys.teku.validator.remote.apiclient.ValidatorApiMethod.SEND_SIGNED_BLOCK;
 
 import java.util.Collections;
+import java.util.Optional;
 import okhttp3.HttpUrl;
 import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
 import tech.pegasys.teku.spec.datastructures.blocks.SignedBeaconBlock;
 import tech.pegasys.teku.validator.api.SendSignedBlockResult;
 import tech.pegasys.teku.validator.remote.apiclient.ValidatorApiMethod;
 import tech.pegasys.teku.validator.remote.typedef.ResponseHandler;
 
 public class SendSignedBlockRequest extends AbstractTypeDefRequest {
 
-  public SendSignedBlockRequest(final HttpUrl baseEndpoint, final OkHttpClient okHttpClient) {
+  private final boolean preferSszBlockEncoding;
+  private boolean retryAsJson = false;
+
+  public SendSignedBlockRequest(
+      final HttpUrl baseEndpoint,
+      final OkHttpClient okHttpClient,
+      final boolean preferSszBlockEncoding) {
     super(baseEndpoint, okHttpClient);
+    this.preferSszBlockEncoding = preferSszBlockEncoding;
   }
 
   public SendSignedBlockResult sendSignedBlock(SignedBeaconBlock signedBeaconBlock) {
     final ValidatorApiMethod apiMethod =
         signedBeaconBlock.getMessage().getBody().isBlinded()
             ? SEND_SIGNED_BLINDED_BLOCK
             : SEND_SIGNED_BLOCK;
+
+    final SendSignedBlockResult result =
+        preferSszBlockEncoding
+            ? sendSignedBlockAsSszOrFallback(apiMethod, signedBeaconBlock)
+            : sendSignedBlockAsJson(apiMethod, signedBeaconBlock);
+    if (!result.isPublished() && retryAsJson) {
+      return sendSignedBlockAsJson(apiMethod, signedBeaconBlock);
+    }
+    return result;
```

Comment:
It's probably worth changing the ?: to a full if and then only checking the retryAsJson inside the branch that sends SSZ.  It works correctly now but feels a little odd that we'd even consider retrying as JSON after having just sent JSON.

Reply:
yeh the logic does work out because of the fact that the retry is only set from a 406... you'll probably want to review the change i make in this spot and see if you're happier with it.